light-client/src/peer_list.rs

```diff
@@ -5,111 +5,169 @@ use crate::{
     types::PeerId,
 };
 
-use contracts::pre;
-use std::collections::HashMap;
+use contracts::{post, pre};
+use std::collections::{HashMap, HashSet};
 
 /// A mapping from PeerIds to Light Client instances.
 /// Keeps track of which peer is deemed the primary peer.
 #[derive(Debug)]
 pub struct PeerList {
-    peers: HashMap<PeerId, Instance>,
+    instances: HashMap<PeerId, Instance>,
     primary: PeerId,
+    witnesses: HashSet<PeerId>,
```

Comment:
witnesses are what is currently called Secondaries in the spec. Will the naming be aligned or is there a particular reason for the difference?

Reply:
I picked `witnesses` to be consistent with the naming in the multi-peer JSON tests, but you are right that we should align them. Let's already merge this and I'll open an issue to track this and to discuss which name we want to keep.